### PR TITLE
Adding members and communities counters + integration + backfilling

### PR DESCRIPTION
**Pull request by royriver23**
## Purpose

Adding new queries to keep track of communities and members' counts. See: https://app.dework.xyz/indiedao/lvl-protocol?taskId=07a1f130-d5a8-4f99-9eaf-309af54eb5a4

## How to setup

- Go to Fauna Dashboard and replace the schema
- Run the newly added shell instructions, see README for more details on how to do this.

## How to test

- You can go into Fauna Dashboard and run the queries yourself from `util/queries.js` 
- Additionally, you may want to actually create members from the app to make sure the counter is updated.

## How to backfill counters in development and production?

- Enable `NEXT_PUBLIC_ENABLE_DATA_MIGRATIONS` in the corresponding environment you want to fix things.
- Call `GET api/backfill-data` endpoint to update counters, you should get a response like this:

```json
{
   "success":true,
   "data":{
      "previousMembersCount":0,
      "previousCommunitiesCount":0,
      "membersCount":0,
      "communitiesCount":2
   }
}
```

- You could also do this locally by setting up the FAUNA_SECRET value to the one from dev or prod.

### CODE REVIEW

File: pages/api/backfill-data.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        communitiesCount: updatedCommunitiesCounter.counter,
+      },
+    })
+  } catch (error) {
+    res.statusCode = 500
+    return res.json({ success: false, error: error.message })
+  }
+}
+
+export default withMethods(['GET'], handler)
```
Comment (mckelvey) [inline pages/api/backfill-data.js:50]: Do we need to put this behind some form of auth? Or is the ENV var intended to be that control?

File: util/fauna.js [hunk 2]
```diff
@@ -22,7 +29,13 @@ const graphQLClient = new GraphQLClient(process.env.NEXT_PUBLIC_FAUNA_URL, {
...
 export const createCommunity = async community => {
   const resp = await graphQLClient.request(CREATE_COMMUNITY_MUTATION, community)
-  return resp.community
+
+  if (resp.community) {
+    await graphQLClient.request(INCREMENT_COMMUNITIES_COUNT)
+    return resp.community
+  }
+
+  return undefined
```
Comment (mckelvey) [inline util/fauna.js:38]: Nice!

File: util/fauna.js [hunk 3]
```diff
@@ -85,9 +104,10 @@ export const updateMemberCache = async ({ id, nftSrc, ens }) => {
   return resp.member
 }
 
-export const getAccessListFirst480 = async () => {
+export const getAccessList = async (size = 480) => {
```
Comment (mckelvey) [inline util/fauna.js:107]: Great handling here.

**Review by mckelvey — APPROVED** [review]
Looking good!

File: pages/api/backfill-data.js (hunk 1 above)
Comment (nicovalencia) [reply pages/api/backfill-data.js:50]: Another way we might want to tackle this is using single-use migration scripts like this: https://github.com/twos-complement/level-protocol-website/tree/develop/scripts/migrations

Comment (nicovalencia) [reply pages/api/backfill-data.js:50]: @royriver23

Comment (royriver23) [reply pages/api/backfill-data.js:50]: @nicovalencia nice idea, how do you generate the file name? is there something like rails migration creation command or so?

Comment (royriver23) [reply pages/api/backfill-data.js:50]: @nicovalencia @mckelvey do you guys know of a way to easily use `import` from scripts? I dug into it and it seems `type: "module"` would work, but it's not doing its work. If not, I can convert the endpoint logic into nodejs `require`, but it requires some rewriting of a bunch of stuff (fauna functions, GQL queries, etc), which I don't know would be worth the effort right, given that we could also get into Fauna and run some queries to backfill, thoughts? 

If I should still convert it, I can invest more time but wanted to check with you, first, cause, this is why this task has been delayed.

**Comment by royriver23** [comment]
Closing in favor of: https://github.com/twos-complement/level-protocol-website/pull/106

**Pull request closed without merging**